pkg/auth/identity.go

```diff
@@ -0,0 +1,70 @@
+package auth
+
+import (
+	"context"
+
+	authapi "k8s.io/api/authorization/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Authorized map[string]bool
+
+type Identity struct {
+	Token   string
+	RestCfg *rest.Config
+	client  k8sclient.Client
+}
+
+func (r *Identity) HasRead(namespaces []string) (Authorized, error) {
+	authorized := Authorized{}
+	err := r.buildClient()
+	if err != nil {
+		return authorized, err
+	}
+	for _, namespace := range namespaces {
+		sar := authapi.SelfSubjectAccessReview{
+			Spec: authapi.SelfSubjectAccessReviewSpec{
+				ResourceAttributes: &authapi.ResourceAttributes{
```

Comment:
Seems like this is asking whether the user has permission to `get` namespaces in the namespace, which doesn't quite make sense to me. Is that a valid way to check for access to a given namespace? I couldn't work that out from the SAR documentation.

Reply:
Yeah so this something weird that took me a lot of trial and error to figure out. Surprisingly specifying the resource name for namespaces doesn't work as you'd expect.

Here's some example output from the oc client:
```
[dymurray@pups mig-controller]$ oc auth can-i get namespaces/dylancantaccess
Warning: resource 'namespaces' is not namespace scoped
yes
[dymurray@pups mig-controller]$ oc auth can-i get namespaces/dylan
Warning: resource 'namespaces' is not namespace scoped
yes
[dymurray@pups mig-controller]$ oc auth can-i get namespaces -n dylancantaccess
Warning: resource 'namespaces' is not namespace scoped
no - no RBAC policy matched
[dymurray@pups mig-controller]$ oc auth can-i get namespaces -n dylan
Warning: resource 'namespaces' is not namespace scoped
yes
```

I don't know if I'm doing it wrong but I think since `namespaces` is a cluster scoped resource that you need to use the `Namespace` param in order to specify which namespace this SAR check is run against. @jortel any thoughts on this?

Reply:
I have seen the same behavior.  May be worth running past David E. at some point but think we should run with it for now.